light_controller/light_controller.py

```diff
@@ -0,0 +1,195 @@
+# -*- coding: utf-8 -*-
+import json
+import os
+import sys
+import time
+
+from light_controller.ir_toy.ir_toy import IR_Toy
+
+__author__ = 'akm'
+
+
+class LightController:
+    """
+    IR Light Controller interface
+
+    Controls several RGB LED globes and strips that use a common 24 button
+    IR remote to change colours.
+
+    They use slightly different protocols you can record the buttons to store
+    in JSON which is loaded by this class.
+
+    :param comPort: Serial device to connect to.
+    """
+    buttonNames = (
+        'brightnessDown', 'brightnessUp', 'off', 'on',
+        'red', 'green', 'blue', 'white',
+        'red2', 'green2', 'blue2', 'flash',
+        'orange', 'cyan', 'purple', 'strobe',
+        'orange2', 'cyan2', 'violet', 'fade',
+        'yellow', 'teal', 'lavender', 'smooth'
+    )
+    _colours = (
+        'red', 'green', 'blue', 'white',
+        'red2', 'green2', 'blue2',
+        'orange', 'cyan', 'purple',
+        'orange2', 'cyan2', 'violet',
+        'yellow', 'teal', 'lavender',
+    )
+    _commands = (
+        'brightnessUp', 'brightnessDown',
+        'flash', 'strobe', 'fade', 'smooth',
+    )
+
+    def __init__(self, comPort, buttons='buttons.json'):
+        self.toy = IR_Toy(comPort)
+        self._buttons = {}
+        self._buttonsFile = buttons
+        self.loadButtons()
+
+    def loadButtons(self):
+        """
+        Load button codes from the file
+        :return: None
+        """
+        if os.path.exists(self._buttonsFile):
+            with open(self._buttonsFile, 'rb') as fp:
+                buttons = json.load(fp)
+                self._buttons = buttons
+
+    def getSettings(self):
+        """
+        Return the IR Toy settings
+        """
+        return self.toy.getSettings()
+
+    def recordButton(self, button, dump=True):
+        """
+        Record a single button press
+
+        :param button: Name of the button
+        :param dump: Whether to save the current button list
+        """
+        self.toy.reset()
+        self.toy.setSamplingMode()
+        print("Press {}".format(button))
+        sys.stdout.flush()
+        irCode = self.toy.receiveSignal()
+        self._buttons[button] = irCode
+        if dump:
+            self.dumpButtons()
+
+    def dumpButtons(self):
+        """
+        Record All The Buttons!
+        Saves the buttons into the configured button file
+        """
+        with open(self._buttonsFile, 'wb') as fp:
+            json.dump(self._buttons, fp)
+
+    def recordButtons(self):
+        """
+        Record button presses from the remote
+        :return: SFA
+        """
+        print("Protocol Version:", self.toy.protocolVersion)
+        for button in sorted(self.buttonNames):
+            self.recordButton(button, dump=False)
+        self.dumpButtons()
+
+    def lightsOn(self):
+        """
+        Turn the lights on
+        :return:
+        """
+        self.toy.transmitCode(self._buttons['on'])
```

Comment:
Instead of hard coded strings, we can initialise as constants and use them here.